### DISCUSSION TITLE
Need to add a feature that saves Only PNG files saves in a sepearate file

### DISCUSSION THREAD

**Discussion by aravindb26** [projectdiscovery/httpx#1149] [General]
### Please describe your feature request:
A feature to save only PNG files in a separate folder after capturing screenshots.

### Describe the use case of this feature:
Currently, the screenshots are saved individually for each domain, which requires manual effort to visit each website. By implementing this feature, all PNG files can be saved in a designated folder, saving time and effort.

### Proposed Solution:
To address this feature request, the following solution can be implemented:

1. Create a new folder to save PNG files: `mkdir png_folder`
2. Use the `find` command to locate PNG files in the "screenshot" directory: `find screenshot -type f -name "*.png" -exec cp {} png_folder/ \;`

By implementing this solution, all PNG files will be copied to the "png_folder," consolidating them in one location.

Please let me know if you need any further information or clarification.

**Comment by Mzack9999** [comment]
@aravindb26 Thank you for raising this issue. The screenshot functionality uses an index to handle multiple input URLs that might refer to the same domain.
Using a hashing mechanism, we can efficiently differentiate between different URLs (ex, with other methods, additional parameters, etc.) while still capturing screenshots for each unique domain. This approach helps us avoid duplicates and ensures we cover all the URLs provided.

If you have any further questions or need more clarification, please feel free to ask. We're here to help!

**Reply by aravindb26** [reply]
@Mzack9999 Thank you for your reply ☺️.

Yeah I understood what you said.

Actually if the output is better, then it would be good for all users like there will be no time consuming taken if user wants to search for 1000s of screenshots in each folder.

So just suggesting if the output will be like all PNG files at one folder.

Thank You

**Comment by bugrakocabay** [comment]
[This feature](https://github.com/projectdiscovery/httpx/pull/1226) might be useful for this issue. :)

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
